### PR TITLE
[REVIEW IS NOT NEEDED] Python3 compatible code

### PR DESCRIPTION
**Pull request by Sae126V**
This PR will be closed soon since #354 have raised breaking this down to multiple commits and NOT disturbing the flow of python2.7.

### CODE REVIEW

File: apel/common/hashing.py [hunk 1]
```diff
@@ -29,21 +26,21 @@
...
+        fp = gzip.open(fname, 'rb')
+        while data != b'':
             # 128kiB buffer
             data = fp.read(131072)
             md.update(data)
-    except IOError: # not a gzipped file
-        fp = open(fname, 'r')
-        while data != '':
+    except (IOError, OSError): # not a gzipped file
+        fp = open(fname, 'rb')
```
[the remark this answers is not in the document]
Comment (gregcorbett) [reply apel/common/hashing.py:42]: could the file be opened with `with` here to ensure the file is always closed?

File: apel/parsers/lsf.py [hunk 2]
```diff
@@ -126,9 +126,10 @@ def parse(self, line):
             data[key] = mapping[key](items)
 
         # Input checking
-        if data['CpuDuration'] < 0:
+        # TODO: Check needed, across codebase: When testing try to compare with schema
```
Comment (gregcorbett) [inline apel/parsers/lsf.py:129]: Open an issue rather than add a TODO to the codebase.

Comment (Sae126V) [reply apel/parsers/lsf.py:129]: I could have given a heads-up. TODO is the activity for my testing with rocky8 and 9 ticket. I will fix those as part of that. I can assure it is not major one. It is doable in very less time, Greg.  I will remove those eventually. Make sense ?

File: bin/retrieve_dns.py [hunk 3]
```diff
@@ -114,14 +114,17 @@ def get_xml(url, proxy):
     '''
     try:
         # Try without a proxy
-        conn = urllib.urlopen(url)
+        conn = urllib.request.urlopen(url)
         dn_xml = conn.read()
         conn.close()
-    except IOError:
+        # TODO: If we are the owners of the url specified then we can catch proper errors than all.
+    except Exception:
```
Comment (gregcorbett) [inline bin/retrieve_dns.py:121]: We probably want to catch more specific exceptions here?

File: apel/db/loader/loader.py [hunk 4]
```diff
@@ -136,7 +136,7 @@ def load_all_msgs(self):
         while self.current_msg:
             if not self._inq.lock(self.current_msg):
                 log.warning("Skipping locked message %s", self.current_msg)
-                self.current_msg = self._inq.next()
+                self.current_msg = next(self._inq)
```
Comment (gregcorbett) [inline apel/db/loader/loader.py:139]: ah okay, this is the Python 3 way of using next - see: https://peps.python.org/pep-3114/

File: apel/common/hashing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gregcorbett) [reply apel/common/hashing.py:42]: but also, not an issue you've introduced so 🤷

File: apel/db/loader/loader.py (hunk 4 above)
Comment (gregcorbett) [reply apel/db/loader/loader.py:139]: though would this break python2.7 support? Does `.next()` still exist in Python 3?

Maybe we need:

```
try:
    self.current_msg = next(self._inq)
except SomeException:
    self.current_msg = self._inq.next()
```

File: bin/retrieve_dns.py (hunk 3 above)
Comment (gregcorbett) [reply bin/retrieve_dns.py:121]: `URLError` reads to me as the base class for all `urllib.request` exceptions: https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError, which would be better to catch then just `Exception`.

> The [urllib.error](https://docs.python.org/3/library/urllib.error.html#module-urllib.error) module defines the exception classes for exceptions raised by [urllib.request](https://docs.python.org/3/library/urllib.request.html#module-urllib.request). The base exception class is [URLError](https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError).

**Comment by tofu-rocketry** [comment]
These are a lot of changes squashed into one commit. It'd be happier if the changes were grouped by category into separate commits.

I thought I'd pointed you at the start I'd made here: https://github.com/apel/apel/compare/dev...tofu-rocketry:apel:python3

Note the separate commits for groups of changes. e.g. separate commits for "Change implicit relative imports to explicit" and "Change .iteritems() to .items() as removed in Py3"

**Comment by Sae126V** [comment]
Sure Adrian. Right from dev activity start, I have not realized I have to do the same breaking into multiple commits. But, I am happy to try and break it into smaller commits.

File: bin/client.py [hunk 5]
```diff
@@ -31,11 +31,7 @@
 import logging.config
 import ldap
 
-try:
-    import ConfigParser
-except ImportError:
-    # Renamed in Python 3
-    import configparser as ConfigParser
+import configparser
```
[the remark this answers is not in the document]
Comment (Sae126V) [reply bin/client.py:34]: Now that we are not disturbing the flow. I have added this in #354  5a6b0c17a65f9d0b363b5b86e3afcbde9a08d5cf. Just to acknowledge, Greg.

File: bin/retrieve_dns.py (hunk 3 above)
Comment (Sae126V) [reply bin/retrieve_dns.py:121]: I have added this in https://github.com/apel/apel/pull/354 https://github.com/apel/apel/commit/5b581a25e47958f5cd9a867dcb80609d4ab72d82. Just to acknowledge, Greg

File: apel/db/loader/loader.py (hunk 4 above)
Comment (Sae126V) [reply apel/db/loader/loader.py:139]: I have added this in https://github.com/apel/apel/pull/354 https://github.com/apel/apel/commit/ef8d4b3312aee2ac17524ad1c566ff3856d64be3 and dd4bd0d3fe02d7a86c33f84c9f8beb117f8d2fa7.

This next() will be handled by future for cross-compatibility(dd4bd0d3fe02d7a86c33f84c9f8beb117f8d2fa7)

**Comment by tofu-rocketry** [comment]
Closing in favour of #354

**Pull request closed without merging**